// DO NOT EDIT: this file is AUTOMATICALLY GENERATED and should not be changed.
syntax = "proto2";

package xls;

// Listing of all XLS IR operations.
enum OpProto {
  OP_INVALID = 0;
  OP_ADD = 1;
  OP_AND = 2;
  OP_AND_REDUCE = 3;
  OP_ASSERT = 4;
  OP_COVER = 5;
  OP_RECEIVE = 6;
  OP_SEND = 7;
  OP_NAND = 8;
  OP_NOR = 9;
  OP_AFTER_ALL = 10;
  OP_ARRAY = 11;
  OP_ARRAY_INDEX = 12;
  OP_ARRAY_SLICE = 13;
  OP_ARRAY_UPDATE = 14;
  OP_ARRAY_CONCAT = 15;
  OP_BIT_SLICE = 16;
  OP_DYNAMIC_BIT_SLICE = 17;
  OP_BIT_SLICE_UPDATE = 18;
  OP_CONCAT = 19;
  OP_COUNTED_FOR = 20;
  OP_DECODE = 21;
  OP_DYNAMIC_COUNTED_FOR = 22;
  OP_ENCODE = 23;
  OP_EQ = 24;
  OP_IDENTITY = 25;
  OP_INVOKE = 26;
  OP_INPUT_PORT = 27;
  OP_LITERAL = 28;
  OP_MAP = 29;
  OP_NE = 30;
  OP_NEG = 31;
  OP_NOT = 32;
  OP_ONE_HOT = 33;
  OP_ONE_HOT_SEL = 34;
  OP_PRIORITY_SEL = 35;
  OP_OR = 36;
  OP_OR_REDUCE = 37;
  OP_OUTPUT_PORT = 38;
  OP_PARAM = 39;
  OP_REGISTER_READ = 40;
  OP_REGISTER_WRITE = 41;
  OP_INSTANTIATION_OUTPUT = 42;
  OP_INSTANTIATION_INPUT = 43;
  OP_REVERSE = 44;
  OP_SDIV = 45;
  OP_SMOD = 46;
  OP_SEL = 47;
  OP_SGE = 48;
  OP_SGT = 49;
  OP_SHLL = 50;
  OP_SHRL = 51;
  OP_SHRA = 52;
  OP_SIGN_EXT = 53;
  OP_SLE = 54;
  OP_SLT = 55;
  OP_SMUL = 56;
  OP_SMULP = 57;
  OP_SUB = 58;
  OP_TUPLE = 59;
  OP_TUPLE_INDEX = 60;
  OP_UDIV = 61;
  OP_UMOD = 62;
  OP_UGE = 63;
  OP_UGT = 64;
  OP_ULE = 65;
  OP_ULT = 66;
  OP_UMUL = 67;
  OP_UMULP = 68;
  OP_XOR = 69;
  OP_XOR_REDUCE = 70;
  OP_ZERO_EXT = 71;
  OP_GATE = 72;
  OP_TRACE = 73;
}
